lib/logstash/codecs/auto_flush.rb

```diff
@@ -11,16 +11,21 @@ def start
     # can't start if pipeline is stopping
     return self if stopped?
     if pending?
-      @task.reset
+      @task.cancel
```

Comment:
@ph - I see what you mean. what happens when the task is executing and we get a start?

Firstly, I just realised a bug: I don't cancel the task when we flush normally this could flush a half built buffer if the lines come along slowly.
Must fix.

Secondly, if the task is running we need to wait on the task future value and then create_task

> Might be good to check if the cancel call was successful, if its the case we start a new task?

What do we do if the cancel call is not successful?

Maybe this works better:
```ruby
  def start
    # can't start if pipeline is stopping
    return self if stopped?

    if pending? && @task.cancel
      create_task
      return
    end
    # maybe we have a timing edge case
    # where pending? was true but cancel failed
    # because the task started running
    if finished?
      create_task
      return
    end
    # else the task is executing
    # wait for task to complete
    # flush could feasibly block on queue access
    @task.value
    create_task
    self
  end
```